daemon/container.go

```diff
@@ -1049,6 +1050,58 @@ func (container *Container) HostConfig() *runconfig.HostConfig {
 	return res
 }
 
+//Populates types.Container struct data
+func (c *Container) WriteAs(container *Container, config *ContainersConfig, newC *types.Container) {
```

Comment:
ok - now I see why 'config' is passed in. I would say that this func shouldn't have that param and it should always fill in the info. Then the caller can remove it from the struct if we really wanted to.  IMO, the API should always return that info and the "ps -s" flag should just control what is displayed, not what is sent across the wire.  But I think that would be a breaking API change so let's just have the caller 'nil' out that field if -s isn't used.